tests/ftest.rb

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+require 'yaml'
+require 'elasticsearch'
+
+LIB_DIR = File.expand_path("#{File.dirname(__FILE__)}/../lib")
+
+def es_client
+  Elasticsearch::Client.new(
+    host: 'http://localhost:9200',
+    user: 'elastic',
+    password: 'changeme',
+  )
+end
+
+def start_stack
+  # starts Mongo and Elasticsearch, adds data in Mongo
+  Dir.chdir(__dir__) {
+    puts('Start the stack')
+    system('make stop-stack')
+    system('make run-stack')
+    puts('Load mongo data')
+    system('make load-data')
+  }
+  puts('Wait for all kids to be up') # can do better (ping in a loop)
+  sleep 60
+end
+
+def prepare_es
+  puts('Wipe existing data')
+  system('curl -X DELETE http://localhost:9200/mongo --user elastic:changeme')
+end
+
+def set_auth
+  config_file = File.expand_path("#{File.dirname(__FILE__)}/../config/connectors.yml")
+
+  body = {
+    "name": 'mongo-connector',
+    "role_descriptors": {
+      "mongo-connector-role": {
+        "cluster": ['all'],
+        "index": [
+          {
+            "names": ['*'],
+            "privileges": ['all']
+          }
+        ]
+      }
+    }
+  }
+  response = es_client.perform_request('POST', '_security/api_key',
+                                       {}, body)
+
+  api_key = response.body['encoded']
+  # copy it into the config
+  config = YAML.load_file(config_file)
+  config['elasticsearch']['api_key'] = api_key.strip
+  File.open(config_file, 'w') { |file| file.write(config.to_yaml) }
+end
+
+def run_sync
+  # late import because the lib import the config when loaded
+  # and we change it in `set_auth`
+  $LOAD_PATH << LIB_DIR
+
+  Dir["#{LIB_DIR}/**/*.rb"].sort.each { |f|
+    next if f.include?('lib/app')
+    next if f.include?('connectors_async')
+
+    absolute_dir = File.dirname(f)
+    relative_dir = absolute_dir.sub(/.*lib\//, '')
+    name = File.basename(f, '.rb')
+    f = File.join(relative_dir, name)
+    require(f)
+  }
+
+  require 'active_support/json'
+  require 'core/elastic_connector_actions'
+  require 'core/connector_settings'
+
+  puts('Create the connector')
+  Core::ElasticConnectorActions.ensure_connectors_index_exists
+  connector_id = Core::ElasticConnectorActions.create_connector('mongo', 'mongo')
```

Comment:
I'm currently [refactoring the CLI](https://github.com/elastic/connectors-ruby/pull/224), and the content index should be created in `create_connector`.

But that is at the bottom of my TO-DO list. I will update the test if it's merged before my refactor PR.

Reply:
sounds good, thanks for the heads up